src/java/org/apache/cassandra/cql3/statements/schema/DropFunctionStatement.java

```diff
@@ -120,9 +121,28 @@ public Keyspaces apply(Keyspaces schema)
         if (!dependentAggregates.isEmpty())
             throw ire("Function '%s' is still referenced by aggregates %s", name, dependentAggregates);
 
+        String dependentMasks = keyspace.tables.stream()
+                                               .filter(table -> hasDependingMask(table, function))
+                                               .map(table -> table.name)
+                                               .collect(joining(", "));
```

Comment:
The approach that is mostly use within the code base is to delegate that type of call to the classes themselves. See `dependentAggregates` bellow.

Reply:
I have just tried to do that with two new `KeyspaceMetadata#tablesUsingFunction(Function)` and `TableMetadata#dependsOn(Function)` methods.